warn/warn.go

```diff
@@ -640,6 +640,52 @@ func depsetIterationWarning(f *build.File, fix bool) []*Finding {
 	return findings
 }
 
+func argumentsOrderWarning(f *build.File, fix bool) []*Finding {
+	argumentType := func(expr build.Expr) int {
+		switch expr := expr.(type) {
+		case *build.UnaryExpr:
+			switch expr.Op {
+			case "**":
+				return 4
+			case "*":
+				return 3
+			}
+		case *build.BinaryExpr:
+			if expr.Op == "=" {
+				return 2
+			}
+		}
+		return 1
+	}
+
+	getComparator := func(args []build.Expr) func(i, j int) bool {
+		return func(i, j int) bool {
+			return argumentType(args[i]) < argumentType(args[j])
+		}
+	}
+
+	findings := []*Finding{}
+	build.Walk(f, func(expr build.Expr, stack []build.Expr) {
+		call, ok := expr.(*build.CallExpr)
+		if !ok {
+			return
+		}
+		comparator := getComparator(call.List)
+		if fix {
+			sort.SliceStable(call.List, comparator)
+			return
+		}
+		if sort.SliceIsSorted(call.List, comparator) {
+			return
+		}
+		start, end := expr.Span()
+		findings = append(findings,
+			makeFinding(f, start, end, "args-order",
+				"Function call arguments should be in the following order: positional, keyword, *args, **kwargs.", true, nil))
```

Comment:
these indents are distracting.  consider making the error message a file-constant so that all of this cleanly fits on one line)

Reply:
I'm not happy about this code either, in each warning function I essentially need to copy the following boilerplate:

    findings := []*Finding{}
    ...
    start, end := ....Span()
    findings = append(findings, makeFinding(f, start, end, "<category>", "<warning text>", true, nil)).

I may make the last two parameters different for some warnings in the future, but right now only the category name and the warning message are different for each function, and the category name can be taken from `FileWarningMap`. Any idea how I can get rid of the boilerplate code? Ideally I'd like to just call something like `newWarning(expr, "<warning text>")`, should I pass such function to each warning function as a callback?